toolkit/tools/internal/pkgjson/pkgjson.go

```diff
@@ -5,13 +5,29 @@ package pkgjson
 
 import (
 	"fmt"
+	"regexp"
 
 	"microsoft.com/pkggen/internal/versioncompare"
 
 	"microsoft.com/pkggen/internal/jsonutils"
 	"microsoft.com/pkggen/internal/logger"
 )
 
+const (
+	packageWithVersionNameIndex       = 1
+	packageWithVersionConditionIndex  = 2
+	packageWithVersionVersionIndex    = 3
+	packageWithVersionExpectedMatches = 4
+)
+
+var (
+	// Regular expression to correctly split a string with the package name and an optional version constraint.
+	// Examples:
+	//	gcc ->       "gcc" "" ""
+	//	gcc=9.1.0 -> "gcc" "=" "1.9.0"
+	packageWithVersionRegex = regexp.MustCompile(`^([^<>=]+)([<>=]+)?(.*)?$`)
```

Comment:
Love it, definitely better. It's great to see more people appreciating the magic of regular expressions. I wanted to keep it simple initially, but this is safer. I've just tweaked it a bit by replacing `(?:[<>][=]?)` with `(?:[<>]=)`, since `(?:[<>=])` is already catching all single-character comparisons. I've also added a few modifications to allow people to use whitespaces without crashing the whole thing:

```
([^><=\s]+)\s*(?:((?:[<>]=)|(?:[<>=]))\s*([^<>=\s]+))?\s*$
```